client/src/components/profile/components/internet.tsx

```diff
@@ -45,52 +40,40 @@ function Info({ message }: { message: string }) {
   return message ? <HelpBlock>{message}</HelpBlock> : null;
 }
 
-class InternetSettings extends Component<InternetProps, InternetState> {
-  static displayName: string;
-  constructor(props: InternetProps) {
-    super(props);
-    const {
-      githubProfile = '',
-      linkedin = '',
-      twitter = '',
-      website = ''
-    } = props;
-
-    this.state = {
-      formValues: { githubProfile, linkedin, twitter, website },
-      originalValues: { githubProfile, linkedin, twitter, website }
-    };
-  }
-
-  toggleEditing = () => {
-    this.props.setIsEditing(false);
-  };
-
-  componentDidUpdate() {
-    const {
-      githubProfile = '',
-      linkedin = '',
-      twitter = '',
-      website = ''
-    } = this.props;
-
-    const { originalValues } = this.state;
+const InternetSettings = ({
+  githubProfile = '',
+  linkedin = '',
+  twitter = '',
+  website = '',
+  t,
+  updateSocials,
+  setIsEditing
+}: InternetProps) => {
+  const [formValues, setFormValues] = useState<Socials>({
+    githubProfile,
+    linkedin,
+    twitter,
+    website
+  });
+  const [originalValues, setOriginalValues] = useState<Socials>({
+    githubProfile,
+    linkedin,
+    twitter,
+    website
+  });
```

Comment:
```suggestion
```
This does faithfully preserve the original implementation, but it was flawed. We don't need to create more state to keep track of the props. We have the props!